concepts/modules/about.md

```diff
@@ -0,0 +1,101 @@
+# Modules
+
+[Modules][modules] in Crystal serves 2 purposes:
+
+The first purpose is to use it as a [namespace][namespace], this is mainly to avoid name clashes.
+But it also forms as a form of grouping code together, this is to make it easier to understand what the code is for.
+
+The second purpose is to use it as a [mixin][mixin], this is to share code to types.
+
+Modules has similarities to classes, but the main difference is that modules can not be instantiated, and thereby don't have instance variables.
+
+To declare a module you use the `module` keyword, followed by the name of the module.
+
+```crystal
+module Foo
+end
+```
+
+## Namespace
+
+A namespace is a way to group code together, this is to avoid name clashes, but also to make it easier to understand what the code is for.
+When wanting to access for example a constant or a class that has been placed inside a namespace you use the `::` operator.
+
+```crystal
+module Foo
+  class Bar
+  end
+end
+
+Foo::Bar.new
+```
+
+## Use it as a mixin
+
+A mixin is a way for another type to be able to use the code inside the module.
+This can be useful when for example wanting multiple classes to have the same "base" functionality.
+Or when wanting to share code between classes that are not related.
```

Comment:
```suggestion
This can be useful when, for example, wanting multiple classes to have the same "base" functionality or when wanting to share code between classes that are not related.
```